packages/flame/lib/src/components/timer_component.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:ui';
+
+import '../../components.dart';
+
+/// A component that uses a [Timer] instance which you can react to when it has finished.
+class TimerComponent extends Component {
+  late final Timer timer;
+  final bool removeOnFinish;
+
+  /// Creates a [TimerComponent]
+  ///
+  /// [period] The period of time in seconds that the tick will be called
+  /// [repeat] When true, this will continue running after [period] is reached
+  /// [autoStart] When true, will start upon instantiation
+  /// [onTick] When provided, will be called everytime [period] is reached. This
+  /// overrides the [onTick] method
+  TimerComponent({
+    required double period,
+    bool repeat = false,
+    bool autoStart = true,
+    this.removeOnFinish = false,
+    VoidCallback? onTick,
+  }) {
+    timer = Timer(
+      period,
+      repeat: repeat,
+      onTick: onTick ?? onTick,
+    );
+
+    if (autoStart) {
+      timer.start();
+    }
+  }
+
+  /// Called everytime the [timer] reached a tick and
+  /// no onStep is provided on the component instantiation
+  /// default implementaiton is a no-op, override this
+  /// to add custom logic.
+  void onTick() {}
```

Comment:
would it be confusing to have this just not called if a lambda is provided on the constructor? what if the default onTick implementation called the lambnda. then people who override can decide if they wanna call super.onTick or not, and there is no magic effects